aea/decision_maker/base.py

```diff
@@ -299,11 +300,9 @@ def execute(self) -> None:
 
         :return: None
         """
-        counter = 0
-        while not self.message_in_queue.empty() and counter < self._max_reactions:
-            counter += 1
+        while not self.message_in_queue.empty():
             message = self.message_in_queue.get_nowait()  # type: Optional[Message]
-            if message is not None:
+            if message is not None and message.protocol_id == INTERNAL_PROTOCOL_ID:
                 self.handle(message)
 
```

Comment:
As a suggestion, I'd put the following code
```python
else:
    logger.warning("Message received by the decision maker is not labelled as 'internal'")
```

or something like that. Might be useful to have for debugging.